src/infra/main/template.yaml

```diff
@@ -144,10 +163,175 @@ Resources:
           - Effect: Allow
             Action:
               - "dynamodb:PutItem"
-              - "dynamodb:Query"
-              - "dynamodb:DeleteItem"
-              - "dynamodb:UpdateItem"
             Resource:
               - Fn::ImportValue: !Sub "${CoreStackName}-AccountStatusTableArn"
       Roles:
         - !Ref AISSampleLambdaHandlerRole
+
+  StatusRetrieverFunction:
+    Type: AWS::Serverless::Function
+    # checkov:skip=CKV_AWS_116: DLQ is not appropriate for a Lambda invoked by an API
+    # checkov:skip=CKV_AWS_115: Function is not configured for function-level concurrent execution Limit
+    Properties:
+      FunctionName: !Sub '${AWS::StackName}-StatusRetrieverFunction'
+      CodeUri: ../../../dist/handlers/status-retriever-handler
+      Timeout: 10
+      Environment:
+        Variables:
+          TABLE_NAME:
+            Fn::ImportValue: !Sub "${CoreStackName}-AccountStatusTableName"
```

Comment:
CoreStackName will need to be updated to ais-core to reference the correct output :)